### PR TITLE
CAL-13 Added NSILI endpoint and updated unit tests to use assertThat.…

### PR DESCRIPTION
**Pull request by troymohl**
#### What does this PR do?
Added a base implementation of the NSILI-endpoint and NSILI-WebEndpoint. 
Updated Unit Tests to use assertThat vs assertTrue, etc.
Updated NSILI MockClient to support testing endpoint.
#### Who is reviewing it?
@bdeining 
@jaymcnallie 
@jlcsmith 
@dcruver 
#### How should this be tested?
JUnit and NSILI MockClient
#### Any background context you want to provide?
This is the stubbed in classes that will be added for implementing the NSILI endpoint to expose DDF data via STANAG 4559. 
#### What are the relevant tickets?
CAL-13
#### Checklist:
- [ ] Documentation Updated
- [ X ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/LibraryImpl.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+            obj = poa.create_reference_with_id(manager_type.getBytes(Charset.forName(ENCODING)),
+                    OrderMgrHelper.id());
+        } else if (manager_type.equals(NsiliManagerType.PRODUCT_MGR.getSpecName())) {
+            ProductMgrImpl productMgr = new ProductMgrImpl();
+            try {
+                poa.activate_object_with_id(manager_type.getBytes(Charset.forName(ENCODING)),
+                        productMgr);
+            } catch (ServantAlreadyActive | ObjectAlreadyActive | WrongPolicy e) {
+                LOGGER.info("Error activating ProductMgr", e);
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/LibraryImpl.java]: LOGGER.error() might be more appropriate for these

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/CatalogMgrImpl.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    @Override
+    public HitCountRequest hit_count(Query aQuery, NameValue[] properties)
+            throws ProcessingFault, InvalidInputParameter, SystemFault {
+        HitCountRequestImpl hitCountRequest = new HitCountRequestImpl();
+
+        try {
+            poa_.activate_object_with_id("hit_count".getBytes(Charset.forName(ENCODING)),
+                    hitCountRequest);
+        } catch (ServantAlreadyActive | ObjectAlreadyActive | WrongPolicy e) {
+            System.out.println("hit_count : Unable to activate hitCountRequest object.");
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/CatalogMgrImpl.java]: Use LOGGER in favor of System.out.print

File: catalog/nsili/nsili-app/src/main/resources/features.xml [hunk 3]
```diff
@@ -21,9 +21,15 @@
         <bundle>mvn:org.codice.alliance.nsili/catalog-nsili-source/${project.version}</bundle>
     </feature>
 
+    <feature name="catalog-nsili-endpoint" install="manual" version="${project.version}"
+             description="DDF Catalog STANAG 4559 Endpoint">
```
Comment (bdeining) [inline catalog/nsili/nsili-app/src/main/resources/features.xml]: Alliance NSILI Endpoint may be more appropriate

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/ProductMgrImpl.java [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    public GetRelatedFilesRequest get_related_files(Product[] products, FileLocation location,
+            String type, NameValue[] properties)
+            throws ProcessingFault, InvalidInputParameter, SystemFault {
+        GetRelatedFilesRequestImpl getRelatedFilesRequest = new GetRelatedFilesRequestImpl();
+
+        try {
+            _poa().activate_object_with_id("get_related_files".getBytes(Charset.forName(ENCODING)),
+                    getRelatedFilesRequest);
+        } catch (ServantAlreadyActive | ObjectAlreadyActive | WrongPolicy e) {
+            System.out.println(
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/ProductMgrImpl.java]: LOGGER

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/OrderMgrImpl.java [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    @Override
+    public OrderRequest order(OrderContents order, NameValue[] properties)
+            throws ProcessingFault, InvalidInputParameter, SystemFault {
+        OrderRequestImpl orderRequest = new OrderRequestImpl();
+
+        try {
+            _poa().activate_object_with_id("order".getBytes(Charset.forName(ENCODING)),
+                    orderRequest);
+        } catch (ServantAlreadyActive | ObjectAlreadyActive | WrongPolicy e) {
+            System.out.println("order : Unable to activate orderRequest object.");
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/OrderMgrImpl.java]: LOGGER

File: catalog/nsili/catalog-nsili-endpoint/pom.xml [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>org.codice.alliance.nsili</groupId>
+        <artifactId>nsili</artifactId>
+        <version>0.1-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <name>Codice :: Alliance :: NSILI :: Endpoint</name>
```
Comment (bdeining) [inline catalog/nsili/catalog-nsili-endpoint/pom.xml]: We can probably remove Codice and just run with `Alliance :: NSILI :: Endpoint`

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/LibraryImpl.java (hunk 1 above)
Comment (troymohl) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/LibraryImpl.java]: Updated to use error vs info logging.

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/CatalogMgrImpl.java (hunk 2 above)
Comment (troymohl) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/CatalogMgrImpl.java]: Updated to use LOGGER.

File: catalog/nsili/nsili-app/src/main/resources/features.xml (hunk 3 above)
Comment (troymohl) [inline catalog/nsili/nsili-app/src/main/resources/features.xml]: Updated the description for both the source and the endpoint to reference Alliance NSILI vs STANAG 4559.

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/ProductMgrImpl.java (hunk 4 above)
Comment (troymohl) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/ProductMgrImpl.java]: Updated to use LOGGER.

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/OrderMgrImpl.java (hunk 5 above)
Comment (troymohl) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/managers/OrderMgrImpl.java]: Updated to use LOGGER.

File: catalog/nsili/catalog-nsili-endpoint/pom.xml (hunk 6 above)
Comment (troymohl) [inline catalog/nsili/catalog-nsili-endpoint/pom.xml]: Updated to remove Codice from the name.

**Comment by troymohl** [comment]
Overall the *Impl files are being worked as separate tickets. The TODOs will be covered and implemented with those tickets.

**Comment by bdeining** [comment]
:+1:

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliWebEndpoint.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/")
+public class NsiliWebEndpoint {
```
Comment (jaymcnallie) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliWebEndpoint.java]: Why not implement `NsiliWeb`?

Comment (troymohl) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliWebEndpoint.java]: I'll update this to implement the NsiliWeb from Common.

**Comment by jaymcnallie** [comment]
:+1:

**Pull request merged**